cli/app.go

```diff
@@ -341,6 +341,145 @@ var app = &cli.App{
 					},
 					Action: DataDeleteTabularAction,
 				},
+				{
+					Name:      "dataset",
```

Comment:
What's the motivation for having this separate from the other dataset commands (like making these subcommands of `viam dataset` vs `viam data dataset`)?

Reply:
I was back and forth on this! I think these commands are fundamentally data commands and our API is also structured as such (`AddBinaryDataToDataset` and `RemoveBinaryDataFromDataset` both live in the data management service), so I felt like the mental model for why these should be subcommands already exists in how we're structuring our APIs. Additionally, I do feel like similarly to add tags / add bboxes, these are operating on the data itself, albeit it is to add the data to a specific dataset. I'm on the fence. What do you think?

Reply:
Ahh I think I didn't realize those rpcs were under the datamanagement service. That totally makes sense to me then!